src/main/scala/org/constellation/concurrency/cuckoo/MutableCuckooFilter.scala

```diff
@@ -0,0 +1,67 @@
+package org.constellation.concurrency.cuckoo
+
+import java.io.{ByteArrayInputStream, ByteArrayOutputStream}
+
+import cats.effect.Sync
+import net.cinnom.nanocuckoo.NanoCuckooFilter
+import org.constellation.schema.snapshot.FilterData
+import cats.syntax.all._
+
+class MutableCuckooFilter[F[_], T](filter: NanoCuckooFilter)(implicit sync: Sync[F]) {
+
+  def insert(item: T)(implicit convert: T => String): F[Boolean] = sync.delay {
+    val converted = convert(item)
+    if (filter.insert(converted))
+      true
+    else {
+      filter.expand()
+      filter.insert(converted)
+    }
+  }
+
+  def delete(item: T)(implicit convert: T => String): F[Boolean] = sync.delay {
+    convert.andThen(filter.delete)(item)
+  }
+
+  def contains(item: T)(implicit convert: T => String): F[Boolean] = sync.delay {
+    convert.andThen(filter.contains)(item)
+  }
+
+  def getFilterData: F[FilterData] =
+    for {
+      capacity <- getCapacity
+      contents <- getFilterContents
+    } yield FilterData(contents, capacity)
+
+  def clear: F[Unit] =
+    for {
+      capacity <- getCapacity
+      contents <- MutableCuckooFilter(capacity).getFilterContents
+      _ <- setFilterContents(contents)
+    } yield ()
+
+  private def getCapacity: F[Long] = sync.delay {
+    filter.getCapacity
+  }
+
+  private def getFilterContents: F[Array[Byte]] = sync.delay {
+    val stream = new ByteArrayOutputStream()
+    try filter.writeMemory(stream)
```

Comment:
What if it fails? Then we get empty stream. I am wondering if we should catch that.

Reply:
If it throws an exception, we're not going to get an empty array, the error is propagated in the F context and this method never returns any value. The stream may had something in it, but it should get GC'ed.